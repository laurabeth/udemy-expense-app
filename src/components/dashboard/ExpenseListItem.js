import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ amount, createdOn, description, id, note }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>Amount: {amount}</p>
      <p>Date: {createdOn.format("ddd, MMM, Do YYYY, LT")}</p>
      {note && <p>Note: {note}</p>}
    </div>
  );
};

export default ExpenseListItem;
