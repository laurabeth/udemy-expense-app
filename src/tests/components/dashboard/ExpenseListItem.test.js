/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../../components/dashboard/ExpenseListItem";
import expensesMock from "../../fixtures/expenses";

describe("expense list item", () => {
  it("renders with data", () => {
    const wrapper = shallow(<ExpenseListItem {...expensesMock[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
