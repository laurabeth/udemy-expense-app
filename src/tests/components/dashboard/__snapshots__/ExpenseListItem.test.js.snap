// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expense list item renders with data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      gum
    </h3>
  </Link>
  <p>
    Amount: 
    $0.50
  </p>
  <p>
    Date: 
    Wed, Dec 31st, 1969
  </p>
</div>
`;
